src/v/cloud_storage/partition_manifest.h

```diff
@@ -80,6 +87,12 @@ class partition_manifest final : public base_manifest {
 
         model::initial_revision_id ntp_revision;
         model::term_id archiver_term;
+        /// Term of the segment (included in segment file name)
+        model::term_id segment_term;
+        /// Offset translation delta at the end of the range
+        model::offset delta_offset_end;
+        /// Version (used by JSON deserializer)
+        segment_meta_version version{segment_meta_version::v1};
```

Comment:
it may be nice if when we bump serde version that we leave a comment about the expectations of the new fields' default value if a new node reads data from an old node. Here it looks like segment_term, delta_offset_end will default to something like std::numeric_limits<int64_t>::min() and version will default to v1. how will _new_ code deal with that scenario?

Reply:
yes, this is expected and there is a compatibility tests for this